integrations/optimum/src/haystack_integrations/components/embedders/optimum_text_embedder.py

```diff
@@ -0,0 +1,200 @@
+from typing import Any, Dict, List, Optional
+
+import torch
+from haystack import component, default_from_dict, default_to_dict
+from haystack.utils import Secret, deserialize_secrets_inplace
+from haystack.utils.hf import HFModelType, check_valid_model, deserialize_hf_model_kwargs, serialize_hf_model_kwargs
+from optimum.onnxruntime import ORTModelForFeatureExtraction
+from transformers import AutoTokenizer
+
+
+class OptimumTextEmbedder:
```

Comment:
It looks like a lot of the code below is essentially the same as the one in the document embedder. Let's move them to a common sub-module and call them instead.

Reply:
I have moved the model loading code into a separate backend sub-module (similar to how we have done with the Sentence Transformers Embedders). It has an `embed()` method that can be used for both the embedders.